hack/netpol/pkg/main/main.go

```diff
@@ -0,0 +1,679 @@
+		reachability1.ExpectAllIngress(Pod("x/a"), false)
+		reachability1.Expect(Pod("y/a"), Pod("x/a"), true)
+		reachability1.Expect(Pod("y/b"), Pod("x/a"), true)
+		reachability1.Expect(Pod("y/c"), Pod("x/a"), true)
+		validate(k8s, reachability1, 80)
+
+		reachability1.PrintSummary(true, true, true)
+	}()
+
+	func() {
+		k8s.CreateOrUpdateNamespace("y", map[string]string{"ns-updated": "true", "ns": "y"})
+		reachability1 := NewReachability(allPods, true)
+		reachability1.ExpectAllIngress(Pod("x/a"), false)
+		reachability1.Expect(Pod("y/a"), Pod("x/a"), true)
+		reachability1.Expect(Pod("y/b"), Pod("x/a"), true)
+		reachability1.Expect(Pod("y/c"), Pod("x/a"), true)
+		validate(k8s, reachability1, 80)
+
+		reachability1.PrintSummary(true, true, true)
+	}()
+
+	func() {
+		k8s.CreateOrUpdateDeployment("z", "zb", 1,
+			map[string]string{
+				"pod":     "b",
+				"updated": "true",
+			} ) // old nginx cause it was before people deleted everything useful from containers
+		// copied from above
+		reachability1 := NewReachability(allPods, true)
+		reachability1.ExpectAllIngress(Pod("x/a"), false)
+		reachability1.Expect(Pod("y/a"), Pod("x/a"), true)
+		reachability1.Expect(Pod("y/b"), Pod("x/a"), true)
+		reachability1.Expect(Pod("y/c"), Pod("x/a"), true)
+
+		// delta... pod z in b has 'updated=true' so its whitelisted.
+		reachability1.Expect(Pod("z/b"), Pod("x/a"), true)
+
+		validate(k8s, reachability1, 80)
+
+		reachability1.PrintSummary(true, true, true)
+	}()
+
+	// NOTE THIS TEST IS COPIED FROM THE ABOVE TEST, only delta being that we
+	// dont have the udpated=true annotation above.
+	func() {
+		k8s.CreateOrUpdateDeployment("z", "zb", 1,
+			map[string]string{
+				"pod": "b",
+				// REMOVE UPDATED ANNOTATION, otherwise identical to above function.
+			} ) // old nginx cause it was before people deleted everything useful from containers
+		// copied from above
+		reachability1 := NewReachability(allPods, true)
+		reachability1.ExpectAllIngress(Pod("x/a"), false)
+		reachability1.Expect(Pod("y/a"), Pod("x/a"), true)
+		reachability1.Expect(Pod("y/b"), Pod("x/a"), true)
+		reachability1.Expect(Pod("y/c"), Pod("x/a"), true)
+
+		// REMOVED DELTA, otherwise identical... this confirms that access is blocked again.
+		validate(k8s, reachability1, 80)
+
+		reachability1.PrintSummary(true, true, true)
+	}()
+
+}
+
+/**
+ginkgo.It("should enforce multiple egress policies with egress allow-all policy taking precedence [Feature:NetworkPolicy]", func() {
+ginkgo.It("should enforce policies to check ingress and egress policies can be controlled independently based on PodSelector [Feature:NetworkPolicy]", func() {
+ginkgo.It("should enforce egress policy allowing traffic to a server in a different namespace based on PodSelector and NamespaceSelector [Feature:NetworkPolicy]", func() {
+*/
+func TestEgressAndIngressIntegration(k8s *Kubernetes, stacked bool) []*Stack {
+	// ingress policies stack
+	builder1 := &NetworkPolicySpecBuilder{}
```

Comment:
can we convert this in a more table format.. 
something like
tables := []struct{
        // struct to define what goes in the builder
        name string
        podSelector map[string]string
        policyType policyType
        ingress  ruleStruct
}{
    {
     // actual values in here for the builder
      name:   "deny-all",
      ....
    },
    {....}
for _, t : range table {
    // build using the "t"
    builder = builder.SetName(t.name).SetPodSelector(t.PodSelector)
    // execute reachability and append to []*Stack
    // you get the drift ;)
}

Reply:
but if it is just PoC then we can also follow up

Reply:
yeah, prefer to follow up on table formatting.